DependencyInjection/HttplugExtension.php

```diff
@@ -207,4 +217,20 @@ private function configureAuthentication(ContainerBuilder $container, array $con
                 ->addArgument(new Reference($authServiceKey));
         }
     }
+
+    /**
+     * Verifies that Puli bundle is installed.
+     *
+     * @param ContainerBuilder $container
+     *
+     * @throws \Exception
+     */
+    private function verifyPuliInstalled(ContainerBuilder $container)
+    {
+        if (false === $container->has('puli.discovery')) {
```

Comment:
Awesome! That would be better and easier to read/understan. 

> This requires to know in what order are the compilers and extensions are loaded

Different bundles extensions loads independently, order is not important. After **all** extensions are loaded Symfony starts with the compiler passes. (The "last minute change before compiling the container") Our compiler passes loads in the order we specify in our HttplugBundle.php file. Since the `puli.discovery` service is defined in an extension we are safe to verify its existens in our compiler pass.